### PR TITLE
Add changelog

### PR DESCRIPTION
**Pull request by tamirkamara**
## What is being addressed

Adding a changelog file and update the pull request template to note about it.

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -1,11 +1,12 @@
-# Resolves issue
+# Resolves HASH_SIGN_FOLLOWED_BY_ISSUE_NUMBER
```
Comment (tanya-borisova) [inline .github/pull_request_template.md:1]: Just curious, is the word "issue" actually a problem? 

In this PR, all I did was to paste the link to my issue after the word "issue" here, and it looks like everything was linked correctly? https://github.com/microsoft/AzureTRE/pull/2359

File: CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
+<!-- markdownlint-disable MD041 -->
+<!-- line format short be: change short description (#pr_numer) -->
+## 0.5.0 (Unreleased)
+
```
Comment (stuartleeks) [inline CHANGELOG.md:4]: Add a "migrations"/"breaking changes" section as per discussion this morning

**Review by stuartleeks — APPROVED** [review]
it will be good to have this in place!

File: .github/pull_request_template.md (hunk 1 above)
Comment (tanya-borisova) [reply .github/pull_request_template.md:1]: Although now I'm not sure, it looks like it might've been linked manually there, although don't think I did it

Comment (tamirkamara) [reply .github/pull_request_template.md:1]: I've seen multiple example where the issue wasn't picked up. always deleted the word issue and then it was ok...

**Pull request merged**